test/engine/SpatialJoinTest.cpp

```diff
@@ -0,0 +1,1957 @@
+  kg += "<node_3> <hasGeometry> <geometry3> .";
+  kg += "<geometry3> <asWKT> \"POINT(-0.11957 51.50333)\" .";
+  // statue of liberty
+  kg += "<node_4> <name> \"Statue of liberty\" .";
+  kg += "<node_4> <hasGeometry> <geometry4> .";
+  kg += "<geometry4> <asWKT> \"POINT(-74.04454 40.68925)\" .";
+  // eiffel tower
+  kg += "<node_5> <name> \"eiffel tower\" .";
+  kg += "<node_5> <hasGeometry> <geometry5> .";
+  kg += "<geometry5> <asWKT> \"POINT(2.29451 48.85825)\" .";
+  return kg;
+}
+
+// this function creates an input as a test set and returns it
+std::string createTestKnowledgeGraph(bool verbose) {
+  auto addPoint = [](string* kg, double lon, double lat) {
+    string name =
+        "Point_" + std::to_string(lon) + "_" + std::to_string(lat) + "";
+    *kg += "";
+
+    *kg += name;
+    *kg += " geo:asWKT Point(";
+    *kg += std::to_string(lon);
+    *kg += " ";
+    *kg += std::to_string(lat);
+    *kg += ")^^geo:wktLiteral .\n";
+
+    double fraction = std::abs(lon - (int)lon);
+    if (fraction > 0.49 && fraction < 0.51) {
+      *kg += name;
+      *kg += " <lon-has-fractional-part> <one-half> .\n";
+    } else if (fraction > 0.33 && fraction < 0.34) {
+      *kg += name;
+      *kg += " <lon-has-fractional-part> <one-third> .\n";
+    } else if (fraction > 0.66 && fraction < 0.67) {
+      *kg += name;
+      *kg += " <lon-has-fractional-part> <two-third> .\n";
+    } else if (fraction < 0.01) {
+      if ((int)lon % 2 == 0) {
+        *kg += name;
+        *kg += " <lon-is-div-by> <two> .\n";  // divisible by two
+      }
+      if ((int)lon % 3 == 0) {
+        *kg += name;
+        *kg += " <lon-is-div-by> <three> .\n";  // divisible by three
+      }
+      if ((int)lon % 4 == 0) {
+        *kg += name;
+        *kg += " <lon-is-div-by> <four> .\n";  // divisible by four
+      }
+      if ((int)lon % 5 == 0) {
+        *kg += name;
+        *kg += " <lon-is-div-by> <five> .\n";  // divisible by five
+      }
+    }
+
+    fraction = std::abs(lat - (int)lat);
+    if (fraction > 0.49 && fraction < 0.51) {
+      *kg += name;
+      *kg += " <lat-has-fractional-part> <one-half> .\n";
+    } else if (fraction > 0.33 && fraction < 0.34) {
+      *kg += name;
+      *kg += " <lat-has-fractional-part> <one-third> .\n";
+    } else if (fraction > 0.66 && fraction < 0.67) {
+      *kg += name;
+      *kg += " <lat-has-fractional-part> <two-third> .\n";
+    } else if (fraction < 0.01) {
+      if ((int)lat % 2 == 0) {
+        *kg += name;
+        *kg += " <lat-is-div-by> <two> .\n";  // divisible by two
+      }
+      if ((int)lat % 3 == 0) {
+        *kg += name;
+        *kg += " <lat-is-div-by> <three> .\n";  // divisible by three
+      }
+      if ((int)lat % 4 == 0) {
+        *kg += name;
+        *kg += " <lat-is-div-by> <four> .\n";  // divisible by four
+      }
+      if ((int)lat % 5 == 0) {
+        *kg += name;
+        *kg += " <lat-is-div-by> <five> .\n";  // divisible by five
+      }
+    }
```

Comment:
Remove/aBstract away the code duplication between `lat` and `long` triples for better understandability.